circle.yml

```diff
@@ -1,25 +1,99 @@
 version: 2.1
+
+orbs:
+  aws-s3: circleci/aws-s3@1.0.11
+
 jobs:
-  build-frontend:
+  frontend-test:
     docker:
       - image: circleci/node:lts
-    working_directory: ~/repo/frontend
+    working_directory: ~/project/frontend
     steps:
-      - checkout
+      - checkout:
+          path: ~/project
       - restore_cache:
-          key: dependencies-{{ checksum "package-lock.json" }}
+          key: frontend-deps--{{ .Branch }}-{{ checksum "package-lock.json" }}
       - run:
           name: Installing frontend dependencies
           command: npm install
       - save_cache:
           paths:
             - node_modules
-          key: dependencies-{{ checksum "package-lock.json" }}
+          key: frontend-deps-{{ checksum "package-lock.json" }}
       - run:
-          name: Running frontend tests
+          name: Run frontend tests
           command: npm test -- --reporters=default --reporters=jest-junit --ci --runInBand
       - store_test_results:
           path: .
+
+  frontend-quality:
+    docker:
+      - image: circleci/node:lts
+    working_directory: ~/project/frontend
+    steps:
+      - checkout:
+          path: ~/project
+      - restore_cache:
+          key: frontend-deps-{{ checksum "package-lock.json" }}
+      - run:
+          name: Installing frontend dependencies
+          command: npm install
+      - save_cache:
+          paths:
+            - node_modules
+          key: frontend-deps-{{ checksum "package-lock.json" }}
+      - run:
+          name: Run frontend quality tests
+          command: npm run lint
+      - store_test_results:
+          path: .
+
+  frontend-build:
+    docker:
+      - image: circleci/node:lts
+    working_directory: ~/project/frontend
+    steps:
+      - checkout:
+          path: ~/project
+      - restore_cache:
+          key: frontend-deps-{{ checksum "package-lock.json" }}
+      - run:
+          name: Installing frontend dependencies
+          command: npm install
+      - save_cache:
+          paths:
+            - node_modules
+          key: frontend-deps-{{ checksum "package-lock.json" }}
+      - run:
```

Comment:
@Agrendalath Here is the `.env` file rendering. It'll only render variables starting with `REACT_APP`, and this allows us to add new variables without editing the file.
React takes care of importing this using the `dotenv` package.

Reply:
Oh, this is cool! I haven't thought about adding them this way. I'll gladly reuse it in Sprints CI :)

Btw. dotenv supports also other files, like `.env.local` and `.env.production`. Just wanted to point it out.

Reply:
Oh, this is cool! I haven't thought about adding them this way. I'll gladly reuse it in Sprints CI :)

Btw. dotenv supports also other files, like `.env.local` and `.env.production`. Just wanted to point it out.

Reply:
That's good to know! I'm going with this implementation for now so that I can reuse the build and deployment scripts and just change the context.